src/rules/__tests__/no-color-css-vars.test.js

```diff
@@ -0,0 +1,207 @@
+const rule = require('../no-color-css-vars')
+const {RuleTester} = require('eslint')
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 'latest',
+    sourceType: 'module',
+    ecmaFeatures: {
+      jsx: true
+    }
+  }
+})
+
+ruleTester.run('no-color-css-vars', rule, {
+  valid: [
+    {
+      code: `{color: 'fg.default'}`
+    },
+    {
+      code: `<circle stroke="var(--color-border-default)" strokeWidth="2" />`
+    },
+    {
+      code: `<circle fill="var(--color-border-default)" strokeWidth="2" />`
+    },
+    {
+      code: `<div style={{ color: 'var(--color-border-default)' }}></div>`
+    }
+  ],
+  invalid: [
+    {
+      code: `<Button sx={{color: 'var(--color-fg-muted)'}}>Test</Button>`,
+      output: `<Button sx={{color: 'fg.muted'}}>Test</Button>`,
+      errors: [
+        {
+          message: 'Replace var(--color-fg-muted) with fg.muted'
+        }
+      ]
+    },
+    {
+      code: `
+        <Box sx={{
+          '&:hover [data-component="copy-link"] button, &:focus [data-component="copy-link"] button': {
+            color: 'var(--color-accent-fg)'
+          }
+        }}>
+        </Box>`,
+      output: `
+        <Box sx={{
+          '&:hover [data-component="copy-link"] button, &:focus [data-component="copy-link"] button': {
+            color: 'accent.fg'
+          }
+        }}>
+        </Box>`,
+      errors: [
+        {
+          message: 'Replace var(--color-accent-fg) with accent.fg'
+        }
+      ]
+    },
+    {
+      code: `<Box sx={{boxShadow: '0 0 0 2px var(--color-canvas-subtle)'}} />`,
+      output: `<Box sx={{boxShadow: '0 0 0 2px canvas.subtle'}} />`,
```

Comment:
`boxShadow` doesn't support referencing variables like this. Instead, you have to pass a function like this:

`boxShadow: theme => `0 0 0 2px ${theme.colors.canvas.subtle}``

Example: https://github.com/primer/react/blob/650a46d5dbf10500d4d9af508c71744665f1bd0d/src/PageLayout/PageLayout.tsx#L141

It might be tricky to update the plugin to support this, so let's check if this is a common pattern before investing the time in updating the plugin.

Reply:
Thank you! That was a real example from dotcom I grabbed. I'll see if there's more than one, though.